[cudnn-frontend] Create a new port with v1.11.0 (devendors this from [libtorch])

luncliff opened the pull request:
Maybe we can name this port [`nvidia-cudnn-frontend`](https://repology.org/project/nvidia-cudnn-frontend/versions) like `nvidia-cutlass`. [`cudnn-frontend`](https://repology.org/project/cudnn-frontend/versions) also exists in https://repology.org/.

- Cherrypick some commits of https://github.com/microsoft/vcpkg/pull/36850 to follow the maintainer guideline
- `libtorch` was using this repository

```log
Computing installation plan...
The following packages are already installed:
    cudnn-frontend:x64-windows@1.11.0
cudnn-frontend:x64-windows is already installed
Total install time: 1.01 ms
cudnn-frontend is header-only and can be used from CMake via:

  find_path(CUDNN_FRONTEND_INCLUDE_DIRS "cudnn_backend_base.h")
  target_include_directories(main PRIVATE ${CUDNN_FRONTEND_INCLUDE_DIRS})   

```

## References

- https://github.com/NVIDIA/cudnn-frontend/releases/tag/v1.11.0
- https://github.com/luncliff/vcpkg-registry/pull/347

## Checklist

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

luncliff committed:
[cudnn-frontend] create a new port with 1.11.0

* https://github.com/NVIDIA/cudnn-frontend/releases/tag/v1.11.0

luncliff commented:
I will see the error logs tonight

JoergAtGithub commented on ports/libtorch/vcpkg.json:
```diff
@@ -1,14 +1,15 @@
   "dependencies": [
     "blas",
     "cpuinfo",
+    "cudnn-frontend",
```
Shouldn't this be a dependency of the optional port feature `cuda`?

luncliff replied:
```suggestion
```

luncliff commented:
Sorry for late check. I couldn't see github for a while. I will update this weekend

luncliff committed:
Update ports/cudnn-frontend/portfile.cmake

Co-authored-by: Victor Romero <romerosanchezv@gmail.com>

luncliff commented on ports/cudnn-frontend/portfile.cmake:
```diff
@@ -0,0 +1,31 @@
+        -DCUDNN_FRONTEND_BUILD_SAMPLES=OFF
+        -DCUDNN_FRONTEND_SKIP_JSON_LIB=OFF
+    MAYBE_UNUSED_VARIABLES
+        CUDNN_FRONTEND_FETCH_PYBINDS_IN_CMAKE
```
My mistake. Will be removed.

luncliff committed:
[cudnn-frontend] remove configure options for python build

luncliff committed:
[libtorch] use cudnn-frontend port

- search cudnn_frontend.h in Dependencies.cmake
- use tensorpipe[cuda] in linux

luncliff committed:
[libtorch] fix vulkan feature build error

luncliff commented:
Hmm... would you review the failure logs? 
Maybe I have to remove `libtorch` port changes in this PR...

### magma
x64-linux, `magma` fails to detect CUDA architecture.
```
-- The CUDA compiler identification is unknown
CMake Error at /vcpkg/downloads/tools/cmake-3.30.1-linux/cmake-3.30.1-linux-x86_64/share/cmake-3.30/Modules/CMakeDetermineCUDACompiler.cmake:266 (message):
  Failed to detect a default CUDA architecture.
``` 

### libtorch[vulkan]

Fails to find `glslc` executable.

### libtorch[llvm]

LLVM header changed. Honestly, I'm not sure I can fix this error properly.
